Update editor.less for .pageLine

Gazook89 opened the pull request:
Change the .pageLine (page break) border from bottom to top, since any text on the same line is part of the following page rather than the preceding page.  Also, `background-color` had a value of `fade( ... )` which I wasn't familiar with, and didn't seem to be doing anything.  Replaced with a slight linear gradient.  I'm not very attached to the gradient.

<img width="867" alt="image" src="https://user-images.githubusercontent.com/58999374/129819008-f49ef929-50e8-460f-9e53-2414df8b159c.png">

Some changes to the file to line up the colons.

I took this up with the thought of maybe someday allowing a toggle to add page number and footer snippets to every new page.

Gazook89 commented:
In the latest commit I basically undid what I originally set out to do, because along the way it was determined that `\page` would only work when alone on a line.  And I wasn't married to the idea.

NOW....this PR is about adding a page number to the `.pageLine` in the editor.  It does this with only CSS using an `:after` pseudo element and css counters, in the same way auto page numbering snippet works.   This may help users navigate long brews in the editor.

This should not be interpreted as a match to a `.pageNumber` snippet, as users can start those however often they like.  However, it *should* match the page counter in the bottom right of the preview window.

<img width="611" alt="image" src="https://user-images.githubusercontent.com/58999374/132967720-4b269e5e-88d4-4643-9400-6950b6957d9a.png">

calculuschild commented on client/homebrew/editor/editor.less:
```diff
@@ -4,41 +4,47 @@
+		height        : 100%;
+		counter-reset : page;
 		.pageLine{
-			background-color : fade(#333, 15%);
```
So I looked it up. `fade` is part of Less. Essentially takes the color and applies an opacity of x% to it.

calculuschild commented:
I do like the page numbering.

I'm not sold on the CSS styling of the line though. Removing all the background color makes it *much* harder to locate the start and end of a page simply because it looks like standard paragraph text with an easy-to-miss underline.

How about something like
```
background: linear-gradient(to bottom, white, #33333328 40%);
border-bottom: #339 solid 1px;
```

This way we keep the underline which I think it important, we keep a distinctive background, but it has just the slightest of gradients at the top to blend into the page above.

This does bring up another question though... I feel like the `\page` separator should actually belong to the page *below*. Like, the divider that created page 2 seems like it should be a part of page 2, not page 1? And scrolling to the *bottom* of the page content to find what page it is doesn't quite click for me. What do you think?

Gazook89 commented:
So to your last question:  That is what I was originally concerned about when I first opened this PR, before I changed direction.  When `\page` was still possible even with other text on the same line (`\page more text here`), it was evident that the following text was being placed on the "second" or new page.   This meant it was more clear to have the solid black border at the *top* of the .newLine.   

However, the change was made that nothing else could appear after \page on the same line, if you wanted to actually create a new page, so this became less relevant.  

In my mind's eye I could see re-allowing text on the same line as \page (only after).   This could play well with the future possibility of #1489, allowing the `.page` element to be targeted by a curly injector right on the same line as `\page`.  For instance, adding a pre-made page layout using a class name:
```
\page {coverPage}
```
Or replacing the default background image:
```
\page {background-image:url(imgur.com)}
```
In both examples the style is being applied to the new page, below \page.  And more to the point for this PR, a page number could be included as done in the latest commit and that page number would be at the "top" of the page in the editor as you prefer.

Of course, the first page has to be considered, since it currently doesn't have a \page line.   So if we do the above, the first page wouldn't have the page number etc.  But what if *all* pages required `\page` at the start, including the first page?  A new document would already come preloaded with \page at the top, along with it's page number.  If the user deletes that first line, it would just show the brewRenderer background with a tip on how to get started (or a random rotating choice of 'tips'):

<img width="700" alt="image" src="https://user-images.githubusercontent.com/58999374/134400363-e1f2e941-a56d-42b6-b1e8-38d4f21b0b17.png">

Overall, this makes the first page consistent with the rest of the pages, allows page numbering from "1" to be displayed at the top of the page, and a convenient place for parent-element injection.  Finally, it is also a good spot for *commented* "page names" for easy reference if we ever introduce code folding by page:

```
\page /* Table of Contents */
\page /* Class Descriptions */
\page /* Equipment */
```

calculuschild commented:
There's still possibility that we can allow content after `\page` on the same line at a later date. Don't let that be the deciding factor, but also keep in mind that is probably *far* down the road because it involves writing a whole separate Markdown extension just for that special case.

We can discuss requiring `\page` at the start (as it is now, the *last* page will be missing a page number) in a separate PR I think, since this one is aimed at just adding page numbers.

I say we move ahead with associating `\page` with the start of a new page. So lets move the border to the top to make it clear `\page` belongs to the page contents below, and let's just leave Page 1 unlabeled for now. It's obvious enough that the top of the document coincides with page 1 for now, and we can address it later after some discussion on the Gitter perhaps.

Gazook89 commented:
Updated this PR to match the change requests.....Page Number starts at '2' now.   Removed the gradient entirely though as I now think that a single line height gradient looks a little silly, and doesn't fit the design of the overall site (gradients are used nowhere else).  

*If* ever we allow `\page` at the start of the document as discussed, the start of the page incrementing will need to be worked out....right now, the first incrementation from `0` is done by counting `.codeEditor`, and then subsequent `.page` elements.  

Does this address #846?  Wasn't sure if we should leave that open and plan for a more explicit page indicator to match the preview side.

calculuschild commented:
> Does this address #846? Wasn't sure if we should leave that open and plan for a more explicit page indicator to match the preview side.

I think it addresses it. I'll try to get this PR merged tonight. Good work!
